manifests/init.pp

```diff
@@ -75,6 +78,7 @@
   if ($debug_level < -1) or ($debug_level > 10) { fail('$debug_level must be a number between -1 and 10') }
   if ($error_level < 0) or ($error_level > 10) { fail('$error_level must be a number between 0 and 10') }
   validate_re($update_cmd, '^(default|security|security-severity:Critical|minimal|minimal-security|minimal-security-severity:Critical)$', '$update_cmd must be either \'default\', \'security\', \'security-severity:Critical\', \'minimal\', \'minimal-security\' or \'minimal-security-severity:Critical\'')
+  if $systemname { validate_string($systemname) }
```

Comment:
The `validate_*` functions are deprecated, besides we want to accept the `undef` value (see previous comment) so you can actually get rid of this.